### DISCUSSION TITLE
More modules

### DISCUSSION THREAD

**Discussion by salarizadi** [Moddable-OpenSource/moddable#1081] [Show and tell]
I'm interested in Moddable SDK, it's really great and amazing, that's why I created a Repository to fix some of the shortcomings of this SDK. From now on you can use the modules I make for free.

Link : https://github.com/salarizadi/moddable-sdk

**Comment by phoddie** [comment]
Welcome, @salarizadi! You've created a nice module for mounting an SD Card on ESP32 so it can be used with the Moddable SDK's File module. Good use of the "XS in C" API to bind your JavaScript class to the native ESP-IDF functions. 

On small tip. Your module uses `xsmcToNumber` to access the configuration properties as numbers. `xsmcToNumber` returns a floating point double which your module then assigns an `int`. You could use `xsmcToInteger` instead to avoid unnecessary integer/double conversion .

It is a little unconventional to have a ZIP file of the module sources in your repository. Maybe you could replace the ZIP file with a directory of its contents? That would make it easier for other developers to use.

Thank you for sharing the module. Can't wait to see what you do next.

**Reply by salarizadi** [reply]
Thank you for your kind opinion, yes, I will definitely extract the files from the zip as soon as possible and also put the help document and how to use it, it is an honor to work with moddable ❤️.

**Reply by salarizadi** [reply]
I did all the things you said and now everyone can use the module very easily.

(https://github.com/salarizadi/moddable-sdk)

**Reply by phoddie** [reply]
@salarizadi – very nice, thank you.

**Comment by stc1988** [comment]
This is awesome work! I  directly include manifest  and confirmed to  use SD module 👍 

```
"include": [	
		{ 
			"git":"https://github.com/salarizadi/moddable-sdk.git", 
			"include":"Modules/SD/sd-module/manifest_spi.json"
		}
	],
```

**Reply by salarizadi** [reply]
I am glad that you were able to use this module, thank you for your opinion, it is a great idea that you used the direct link of GitHub, I had not thought of it. 🌹

**Reply by phoddie** [reply]
@stc1988 – very cool use of the (pretty new) git include feature in manifests.
